server/controllers/userController.js

```diff
@@ -0,0 +1,56 @@
+const bcrypt = require("bcrypt");
+const database = require("../services/database");
+
+// functions
+module.exports = {
+  /**
+   * Adds a new user to the database, includes password hashing.
+   * @param {String} displayName
+   * @param {String} email
+   * @param {String} password
+   */
+  addNewUser: function (displayName, email, password) {
```

Comment:
i need to update this function before merging into main - i wrote it with a different authentication strategy in mind